sway/config/bar.c

```diff
@@ -236,9 +236,13 @@ static void invoke_swaybar(struct bar_config *bar) {
 			setenv("WAYLAND_SOCKET", wayland_socket_str, true);
 
 			// run custom swaybar
-			char *const cmd[] = {
-					bar->swaybar_command ? bar->swaybar_command : "swaybar",
-					"-b", bar->id, NULL};
+			const char *swaybarcmd =
+				bar->swaybar_command ? bar->swaybar_command : "exec swaybar";
+			int cmdlen = snprintf(NULL, 0, "%s -b %s", swaybarcmd, bar->id);
+			char *full_cmd = malloc(cmdlen);
+			sprintf(full_cmd, "%s -b %s", swaybarcmd, bar->id);
+			full_cmd[cmdlen] = '\0';
```

Comment:
You're corrupting the heap here. The malloc is not large enough for the null byte here.